pkg/certificate/certificate.go

```diff
@@ -0,0 +1,29 @@
+package certificate
+
+import (
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+
+	corev1listers "k8s.io/client-go/listers/core/v1"
+)
+
+func GetCert(c corev1listers.SecretLister, name, namespace string) (*x509.Certificate, *rsa.PrivateKey, error) {
+	caSecret, err := c.Secrets(namespace).Get(name)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	block1, _ := pem.Decode(caSecret.Data["tls.crt"])
+	certs, err := x509.ParseCertificates(block1.Bytes)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	block2, _ := pem.Decode(caSecret.Data["tls.key"])
```

Comment:
same case with `tls.crt`, and rename the block1 and block2 to crtBlock and keyBlock (anyway we need a meaning name)